### PR TITLE
refactor: split  into two type migration: live-migration and cold migration

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: Michael Wan <zirenwan@gmail.com>

The core migration logic may be a little bit complicated, So I refactor the migration into two type: `live-migrate` and `cold-migrate`.

The total migration still has three steps: `PreMigrate`, `Migrate` and `PostMigrate`, but the `live-migrate` and `cold-migrate` do different things in the steps.

Make the code more simplify also helps to add test cases.

### CODE REVIEW

File: migrator/cold_migrate.go [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+	}
+	pouchVolumes, err := convertor.ToVolumes(dockerVolumes)
+	if err != nil {
+		return err
+	}
+
+	var (
+		// volumeRefs to count volume references
+		volumeRefs    = map[string]string{}
+		pouchHomeDir  = getPouchHomeDir(cm.dockerHomeDir)
```
Comment (zhuangqh) [inline migrator/cold_migrate.go]: `getPouchHomeDir` could be treated as a convertor too. what about moving it to `convertor.ToPouchHomeDir`?
what's more, why don't you just store the pouchHomeDir inside the migrator, it is unnecessary to generate every time....

File: migrator/migrator_utils.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	return nil
+}
+
+func migrateNetworkFile(dockerHomeDir, pouchHomeDir string) error {
+	// Copy network db file
+	var (
+		dbName    = "local-kv.db"
+		dbFile    = path.Join(dockerHomeDir, "network/files", dbName)
```
Comment (zhuangqh) [inline migrator/migrator_utils.go]: split `network/files` into two items? `path.Join(dockerHomeDir, "network", "files", dbName)`

**Comment by zhuangqh** [comment]
Each migrator works fine if there is no container on the host at the very beginning. good

File: migrator/migrator_utils.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	if _, err := os.Stat(newDBDir); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(newDBDir, 0666); err != nil {
+			return fmt.Errorf("failed to prepare network dir: %v", err)
+		}
+	}
+
+	// clear the network directory, in case dirty data remained.
+	// should never occur errors here
+	if err := os.RemoveAll(newDBFile); err != nil {
+		return fmt.Errorf("failed to delelte old network db file: %v", err)
```
[the remark this answers is not in the document]
Comment (HusterWan) [reply migrator/migrator_utils.go]: good catch

File: migrator/cold_migrate.go (hunk 1 above)
Comment (HusterWan) [reply migrator/cold_migrate.go]: `pouchHomeDir` can get from `dockerHomeDir`, so I think no need to store this parameters

File: migrator/migrator_utils.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (HusterWan) [reply migrator/migrator_utils.go]: good catch

File: migrator/migrator.go [hunk 4]
```diff
@@ -2,705 +2,149 @@ package migrator
...
-	if err == nil { // container already exist
-		if err := p.containerd.DeleteContainer(ctx, meta.ID); err != nil {
-			return fmt.Errorf("failed to delete already existed containerd container %s: %v", meta.ID, err)
-		}
-	} else if !errdefs.IsNotFound(err) {
-		return fmt.Errorf("failed to get containerd container: %v", err)
-	}
-
-	return p.containerd.NewContainer(ctx, meta.ID)
+// Migrate  dose migrate procedures
```
[the remark this answers is not in the document]
Comment (HusterWan) [reply migrator/migrator.go]: good catch

File: migrator/cold_migrate.go (hunk 1 above)
Comment (HusterWan) [reply migrator/cold_migrate.go]: `getPouchHomeDir could be treated as a convertor too. what about moving it to convertor.ToPouchHomeDir?`

`convertor` is used to convert the container meta json, not the config file. So I don't think we should mv the `getPouchHomeDir` to `convertor.ToPouchHomeDir`. WDYT?

File: migrator/migrator_utils.go (hunk 2 above)
Comment (HusterWan) [reply migrator/migrator_utils.go]: Updated

**Comment by zhuangqh** [comment]
LGTM good job👍

**Pull request merged**